rclcpp/include/rclcpp/node.hpp

```diff
@@ -892,11 +892,11 @@ class Node : public std::enable_shared_from_this<Node>
 
   RCLCPP_PUBLIC
   rclcpp::Clock::SharedPtr
-  get_clock();
+  get_clock() const;
```

Comment:
Since `Clock` offers non-const methods like `get_clock_handle` which allows you to modify the clock I would argue that a `get_clock() const` should probably only return a const shared pointer (which could be added as parallel API beside the existing non-const method)?

Reply:
To restate that: you're suggesting both of these

```
rclcpp::Clock::ConstSharedPtr get_clock() const;
rclcpp::Clock::SharedPtr get_clock();
```

Correct? I can get behind that

Reply:
Correct.

Reply:
Correct.

Reply:
Done.

Reply:
I would think  the same is necessary in the other APIs as well.

Reply:
I thought as well, I was a little surprised CI passed without changing `NodeClock` and `NodeClockInterface`.